shared/src/main/scala/pl/metastack/metatime/PatternTime.scala

```diff
@@ -0,0 +1,66 @@
+package pl.metastack.metatime
+
+case class PatternTime(parts: Seq[Time => String] = Nil) {
+  def string(s: String): PatternTime = add(_ => s)
+
+  def dash : PatternTime = string("-")
```

Comment:
You can add a type parameter (`T <: Pattern`) to `Pattern`. In `PatternDateTime` extend from `Pattern[PatternDateTime]`. Then, make the method `add` abstract and return `T`. You will need a second type parameter which you initialise with `DateTime`.